src/serialization_support.cpp

```diff
@@ -684,17 +684,33 @@ rosidl_dynamic_typesupport_fastrtps_init_serialization_support_interface(
 
 
   // DYNAMIC TYPE NESTED
+  serialization_support_interface->dynamic_data_get_nested_data =
+    fastrtps__dynamic_data_get_nested_data;
+
+  serialization_support_interface->dynamic_data_set_nested_data =
+    fastrtps__dynamic_data_set_nested_data;
+
+  serialization_support_interface->dynamic_data_insert_nested_data_copy =
+    fastrtps__dynamic_data_insert_nested_data_copy;
+
+  serialization_support_interface->dynamic_data_insert_nested_data =
+    fastrtps__dynamic_data_insert_nested_data;
+
+  // Deprecated.
   serialization_support_interface->dynamic_data_get_complex_value =
-    fastrtps__dynamic_data_get_complex_value;
+    fastrtps__dynamic_data_get_nested_data;
 
+  // Deprecated.
   serialization_support_interface->dynamic_data_set_complex_value =
-    fastrtps__dynamic_data_set_complex_value;
+    fastrtps__dynamic_data_set_nested_data;
 
+  // Deprecated.
   serialization_support_interface->dynamic_data_insert_complex_value_copy =
-    fastrtps__dynamic_data_insert_complex_value_copy;
+    fastrtps__dynamic_data_insert_nested_data_copy;
 
+  // Deprecated.
   serialization_support_interface->dynamic_data_insert_complex_value =
-    fastrtps__dynamic_data_insert_complex_value;
+    fastrtps__dynamic_data_insert_nested_data;
```

Comment:
If my suggestion [here](https://github.com/ros2/rosidl_dynamic_typesupport/pull/13#pullrequestreview-2528837032) is followed, we could then remove these lines